### PR TITLE
Sh/package version install

### PR DESCRIPTION
**Pull request by shetzel**
Updates the plugin with the `PackageVersion` refactoring.  The `PackageVersion` class now has more meaningful instance and static functions, rather than a group of utility functions on a class.

NOTE: requires https://github.com/forcedotcom/packaging/pull/63 to be merged first.

@W-11819142@

For testing: link the `sh/packageVersionInstall` branch of the packaging library to the plugin, then modify the tsconfig.json files compiler options to force resolution of core and jsforce to the same version.
```
"baseUrl": ".",
"paths": {
  "@salesforce/core": ["node_modules/@salesforce/core"],
  "jsforce": ["node_modules/jsforce"]
}
```

### CODE REVIEW

File: src/commands/force/package/beta/version/create.ts [hunk 1]
```diff
@@ -189,10 +189,14 @@ export class PackageVersionCreateCommand extends SfdxCommand {
...
-    const pv = new PackageVersion({ project: this.project, connection: this.hubOrg.getConnection() });
     this.ux.startSpinner(messages.getMessage('requestInProgress'));
-    const result = await pv.create(
-      { ...this.flags, ...{ packageId } },
+    const result = await PackageVersion.create(
+      {
+        connection: this.hubOrg.getConnection(),
+        project: this.project,
+        ...this.flags,
+        ...{ packageId },
```
Comment (WillieRuemmele) [inline src/commands/force/package/beta/version/create.ts:198]: NIT: `...{ }` for one item effectively cancels out, we should be able to pass as just `packageId`

File: src/commands/force/package/beta/version/create/list.ts [hunk 2]
```diff
@@ -34,8 +34,8 @@ export class PackageVersionCreateListCommand extends SfdxCommand {
   };
 
   public async run(): Promise<PackageVersionCreateRequestResult[]> {
-    const pv = new PackageVersion({ connection: this.hubOrg.getConnection(), project: undefined });
-    const results = await pv.createdList({ ...this.flags });
+    const connection = this.hubOrg.getConnection();
+    const results = await PackageVersion.createdList(connection, { ...this.flags });
```
Comment (WillieRuemmele) [inline src/commands/force/package/beta/version/create/list.ts:38]: an even smaller NIT: sometimes we're setting a `const connection = this.hubOrg.getConnection` and other times we're passing `this.hubOrg.getConnection` right into the library

**Review by WillieRuemmele — APPROVED** [review]

File: src/commands/force/package/beta/version/create/list.ts (hunk 2 above)
Comment (shetzel) [reply src/commands/force/package/beta/version/create/list.ts:38]: I think most cases we just pass `this.hubOrg.getConnection()` but in this function the connection was being reused several times so it made sense to assign to a const.

File: src/commands/force/package/beta/version/create.ts (hunk 1 above)
Comment (shetzel) [reply src/commands/force/package/beta/version/create.ts:198]: Good catch.  I'll change it.

File: src/commands/force/package/beta/version/create/list.ts (hunk 2 above)
Comment (WillieRuemmele) [reply src/commands/force/package/beta/version/create/list.ts:38]: I figured I probably missed that in the GH diff view, it's really not a big deal :)

**Pull request merged**